03 Writing Algorithms/12 Universes/03 Equity/01 Liquidity Universes/99 Examples.html

```diff
@@ -1,43 +1,140 @@
     {
-        // Define the universe settings and constituents.
-        UniverseSettings.Resolution = Resolution.Daily;
-        AddUniverse(Universe.DollarVolume.Top(100));
+        // Using asynchronous universe settings can efficiently use the computational resources.
+        UniverseSettings.Asynchronous = true;
+        // Minute resolution data can have orders filled more realistic and avoid stale fill.
+        UniverseSettings.Resolution = Resolution.Minute;
+        // Add top 10 dollar volume universe to trade on the most liquid equities, which has the most informed information from market activities.
+        AddUniverse(Universe.DollarVolume.Top(10));
     }
 }</pre>
-	<pre class="python">class DailyLiquidUniverseAlgorithm(QCAlgorithm):
+	<pre class="python">class MinuteLiquidUniverseAlgorithm(QCAlgorithm):
 
-    def initialize(self):
-        # Define the universe settings and constituents.
-        self.universe_settings.resolution = Resolution.DAILY
-        self.add_universe(self.universe.dollar_volume.top(100))</pre>
+    def initialize(self) -&gt; None:
+        # Using asynchronous universe settings can efficiently use the computational resources.
+        self.universe_settings.asynchronous = True
+        # Minute resolution data can have orders filled more realistic and avoid stale fill.
+        self.universe_settings.resolution = Resolution.MINUTE
+        # Add top 10 dollar volume universe to trade on the most liquid equities, which has the most informed information from market activities.
+        self.add_universe(self.universe.dollar_volume.top(10))</pre>
 </div>
 
-<h4>Example 2: Asynchronous Liquid Universe</h4>
-<p>The following example <a href='/docs/v2/writing-algorithms/universes/settings#09-Asynchronous-Selection'>asynchronously selects</a> the 1,000 most liquid stocks:</p>
+<h4>Example 2: Low Bollinger Bands Liquid Universe</h4>
+<p>The following example <a href='/docs/v2/writing-algorithms/universes/settings#09-Asynchronous-Selection'>asynchronously selects</a> the 500 most liquid stocks. Among them, select the ones with 20-day Bollinger Bands below 2 standard deviations. Place <a href="">market-on-open orders</a> and bet them to revert to mean.</p>
 <div class="section-example-container">
 	<pre class="csharp">public class AsynchronousLiquidUniverseAlgorithm : QCAlgorithm
 {
+    private Dictionary&lt;Symbol, BollingerBands&gt; _bbBySymbols = new();
+    
     public override void Initialize()
     {
-        // Define the universe settings and constituents.
+        // Using asynchronous universe settings can efficiently use the computational resources.
         UniverseSettings.Asynchronous = true;
-        AddUniverse(Universe.DollarVolume.Top(1000));
+        // Subscribe to daily data for speed, accuracy is not affected since the trading frequency is only at daily market open.
+        UniverseSettings.Resolution = Resolution.Daily;
+        AddUniverse(Selection);
```

Comment:
The idea is to use 
```csharp
AddUniverse(Universe.DollarVolume.Top(500));
```
since we want to this pre-build universe here.
Then we add `BB` in `OnSecuritiesChanged`. It's not as efficient, but it's an option. We can use this example in "Fundamental Universe" to show members how to do it differently, then we link the examples.